include/rtsan_standalone/rtsan_standalone.h

```diff
@@ -84,6 +89,9 @@ class ScopedDisabler {
   ScopedDisabler(const ScopedDisabler &);
   ScopedDisabler &operator=(const ScopedDisabler &);
 #endif // __cplusplus >= 201103L
+
+#define __RTSAN_MARK_FUNCTION_BLOCKING()  \
+  __rtsan_notify_blocking_call(__func__)
```

Comment:
So __func__ is standard since c99, and has been standardized in C++ since C++11.

The major compilers (gcc and clang) have supported this for a very long time, including support for it pre-c++11.

I think this macro is medium useful, but it will not strictly work in ALL cases (old compiler, running an old standard), so I would be amenable to keeping it or taking it out.

If we were to leave this in, and this failed on someone's system, they could easily just use the `__rtsan_notify_blocking_call("manualname")` directly. Although, that being said it would be trivial for 99% of users to do this as well with `__rtsan_notify_blocking_call(__func__)`. 

So in total:

¯\_(ツ)_/¯

I defer to you



[[source](https://stackoverflow.com/a/4384825)]

Reply:
Cool! I think let's keep `__func__`, but I have a couple of another suggestion for the macro - the word "mark" makes me think of adding attributes rather than calling functions. I'd suggest maybe we just go with something simple `__RTSAN_NOTIFY_BLOCKING` - WDYT?

Also, does this macro require a semicolon after it at the call site? I'm aware there are some [tricks to making this required](https://stackoverflow.com/questions/163365/how-do-i-make-a-c-macro-behave-like-a-function), but I'm not very familiar with macro magic.

Reply:
Yep, fortunately that macro magic can all be avoided if you call a function in your macro, and omit the last semicolon, this forces the user to add one.

That's what we are doing here:

#define __RTSAN_MARK_FUNCTION_BLOCKING()  \
  __rtsan_notify_blocking_call(__func__) <---- no semicolon, when expanded, wont compile without one
  
I am on the fence about changing to "notify". let me ponder it a bit and get back to you

Reply:
I think I'm in favor of your suggestion - mostly because it decreases introduction of new terms. I think there is value in keeping our terminology as "small surface area" as possible

Reply:
Cool - sounds good!